sandbox/apparmor/notify/message.go

```diff
@@ -0,0 +1,418 @@
+package notify
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/snapcore/snapd/arch"
+)
+
+// Message fields are defined as raw sized integer types as the same type may be
+// packed as 16 bit or 32 bit integer, to accommodate other fields in the
+// structure.
+
+// MsgHeader is the header of all apparmor notification messages.
+//
+// The header encodes the size of the entire message, including variable-size
+// components and the version of the notification protocol.
+type MsgHeader struct {
+	// Length is the length of the entire message, including any more
+	// specialized messages appended after the header and any variable-length
+	// objects referenced by any such message.
+	Length uint16
+	// Version is the version of the communication protocol.
+	// Currently version 2 is implemented by the demo kernel based on Linux 5.7.7,
+	// but version 3 was used in the past.
```

Comment:
Is version 2 still being used?

Reply:
No. Only version 3 atm. We will have to transition to a version 4 at some point to add some new information.

Reply:
Thanks, I'll make the change. Is there a spec I could reference with differences between the versions?